apps/weather_service/lib/weather_service.ex

```diff
@@ -12,8 +14,13 @@ defmodule WeatherService do
       :world
 
   """
-  @spec get_weather(map) :: {:ok, map} | {:error, atom}
-  def get_weather(params) do
-    {:ok, %{id: 1}}
+  @spec get_weather(String.t, String.t) :: {:ok, String.t} | {:error, atom}
+  def get_weather(correlation_id, city) do
+    Logger.info("Getting Weather: #{correlation_id}")
+    with {:ok, weather} <- Weather.weather(%{city: city}, correlation_id) do
+      {:ok, weather}
+    else
+      err -> err
```

Comment:
The `else` statement could be removed here, at least you want to personalize the response error